[Kumar Prabhat] iP

dedsecrattle opened the pull request:
# Jivox ChatBot

All you need to do is -
- Download it from [here](https://github.com/dedsecrattle/ip/releases/tag/A-Jar)
- and Run the Application through Java using `java -jar jivox.jar`

1. Todo, Deadline and Event Tasks
2. Mark , Unmark and find tasks using keyword
3. See deadlines on certain *dates*
4. ~FAST~ SUPER FAST to use

And its **Free** to Use

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```


- [x] Managing Tasks
- [x] Finding tasks using Keyword
- [x] Appealing UI

:smile:

> In Programming , No News is a Good News

[My Github](https://github.com/dedsecrattle)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

dedsecrattle committed:
Level-0 Add Jivox and Delete Duke

dedsecrattle committed:
Level-1 Add Echo

dedsecrattle committed:
Level-2 Implement Add and List

dedsecrattle committed:
Level-3 Add mark and unmark

dedsecrattle committed:
Level-4 Add Todo, Event, Deadline

dedsecrattle committed:
Level-5 Add Exception Handling

dedsecrattle committed:
Level-6 Add Delete functionality

dedsecrattle committed:
A-TextUiTesting Done

dedsecrattle committed:
A-enums Complete

dedsecrattle committed:
Add Local Database to store tasks

dedsecrattle committed:
Implement Level-7

dedsecrattle committed:
Fix Database

dedsecrattle committed:
Merge branch 'branch-Level-7'
Merged Branch-level-7 to Master

dedsecrattle committed:
Add DateTime

dedsecrattle committed:
Merge branch 'branch-Level-8'

dedsecrattle committed:
Abstracted out New Classes

dedsecrattle committed:
A-Packages Done

dedsecrattle committed:
Merge branch 'add-gradle-support'

dedsecrattle committed:
Added Gradle to Project

dedsecrattle committed:
Add JUnit tests

dedsecrattle committed:
Add JUnit tests more

dedsecrattle committed:
Update .gitignore

dedsecrattle committed:
Add A-Jar

dedsecrattle committed:
Add Find functionality

dedsecrattle committed:
Fix Checkstyle and Add Gradle checkstyle

dedsecrattle committed:
Add JavaDoc

dedsecrattle committed:
Merge branch 'branch-A-JavaDoc'

dedsecrattle committed:
Merge branch 'branch-A-CodingStandard'

dedsecrattle committed:
Checkstyle Fix

dedsecrattle committed:
Merge branch 'branch-Level-9'

dedsecrattle committed:
Add Find functionality Level-9

dedsecrattle committed:
Fix Find Functionality - Complete Level-9

dedsecrattle committed:
Fix checkstyle and Update JavaDoc

dedsecrattle committed:
Update Gradle

dedsecrattle committed:
Add JavaFX

dedsecrattle committed:
Merge branch 'branch-Level-10'

dedsecrattle committed:
Fix Checkstyle

jxunze commented on src/main/java/jivox/Ui.java:
```diff
@@ -0,0 +1,182 @@
+    /**
+     * Shows a divider.
+     *
+     * @return
```
return is empty even though function returns a string

jxunze commented on src/test/java/jivox/DeadlineTest.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    @Test
+    public void testToString() {
```
Do look into the naming conventions for junit test. Testname should express a specific requirement. Here is a [link](https://medium.com/@stefanovskyi/unit-test-naming-conventions-dd9208eadbea) for reference!

jxunze commented on src/test/java/jivox/EventTest.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    @Test
+    public void testGetType() {
```
ditto

jxunze commented on src/test/java/jivox/TaskListTest.java:
```diff
@@ -0,0 +1,58 @@
+    }
+
+    @Test
+    public void testDelete() {
```
ditto

jxunze commented on src/test/java/jivox/TodoTest.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    @Test
+    public void testGetType() {
```
ditto

jxunze commented on src/main/java/jivox/Jivox.java:
```diff
@@ -0,0 +1,233 @@
+        jivox.run();
+    }
+
+}
```
Javadocs seem to miss on return portion.

jxunze reviewed (commented):
Generally clean and readable code, less some minor naming conventions issues for the test cases. Good job!

dedsecrattle committed:
Add Streams

dedsecrattle committed:
Merge branch 'branch-A-Streams'

dedsecrattle committed:
Fix CodeQuality

Following are the Changes -
- Removed reductant saving of tasks at every command
- refactored the code
- Resolved some Code Quality issues

dedsecrattle committed:
Merge branch 'branch-A-CodeQuality'

dedsecrattle committed:
Add BCD-Extension

dedsecrattle committed:
Add onClose Event Handle

Save the tasklist if the user closes the application
forcefully

dedsecrattle committed:
Merge pull request #1 from dedsecrattle/branch-A-CodeQuality

Add onClose Event Handle

dedsecrattle committed:
Add Assertions

dedsecrattle committed:
Implement A-Assertions

Following Changes made -
- Add assertions in 'Jivox.java', 'DatabaseHandler.java'
- Removed Commented Code

dedsecrattle committed:
Add A-CI

-Setup Github Actions for Continous Integration
-Used Gradle to check

dedsecrattle committed:
Fix Checkstyle

- Fixed Checkstyle
- Remove commented code
- Add javadoc

dedsecrattle committed:
Fix Checkstyle

dedsecrattle committed:
Add C-Tagging

dedsecrattle committed:
Implement BCD-Extension

Following Extensions Added -
1. B-Duplicated - No Duplicate Task can be added
2. C-Tagging - Tasks can be tagged with a Keyword

dedsecrattle committed:
Fix Checkstyle

- Checkstyle issues fixe
- Commented code remove
- Refactor code

dedsecrattle committed:
Complete Week-5 iP Tasks

dedsecrattle committed:
Better GUI

dedsecrattle committed:
GUI Improved

dedsecrattle committed:
Fix Checkstyle

dedsecrattle committed:
Merge pull request #3 from dedsecrattle/A-MoreBeautifulGUI

A more beautifulGUI

dedsecrattle committed:
Update README.md

Added UserGuide for the Bot

dedsecrattle committed:
Update UI

dedsecrattle committed:
Add Jivox Exceptions Classes

dedsecrattle committed:
Fix Checkstyle

dedsecrattle committed:
Merge pull request #4 from dedsecrattle/A-MoreExceptions

Add Jivox Exceptions Classes

dedsecrattle committed:
Update Exceptions

dedsecrattle committed:
Improve Code Quality

dedsecrattle committed:
Fix Emoji

dedsecrattle committed:
Add Followings :-

1- Ui.png added to docs
2- More Exceptions Handled

dedsecrattle committed:
Fix Checkstyle

dedsecrattle committed:
Update Font

dedsecrattle committed:
Fix Exit Bug

dedsecrattle committed:
Fix jUnit Test

dedsecrattle committed:
Fix DatabaseHandlerTest

dedsecrattle committed:
Update DialogBox.fxml

dedsecrattle committed:
Fix Bug in Delete Command
